omnigibson/systems/micro_particle_system.py

```diff
@@ -1491,42 +1353,173 @@ def cm_customize_particle_material(cls):
                 customize_particle_material(mat=cls._material)
 
         # Add to any other params specified
-        kwargs["_register_system"] = cp_register_system
-        kwargs["particle_contact_offset"] = cp_particle_contact_offset
-        kwargs["particle_density"] = cp_particle_density
         kwargs["_material_mtl_name"] = cp_material_mtl_name
         kwargs["_customize_particle_material"] = cm_customize_particle_material
 
         # Create and return the class
-        return subclass_factory(name=f"{fluid_name}System", base_classes=FluidSystem, **kwargs)
+        return super().create(
+            name=name,
+            particle_contact_offset=particle_contact_offset,
+            particle_density=particle_density,
+            **kwargs,
+        )
 
+def customize_particle_material_factory(attr, value):
+    def func(mat):
+        setattr(mat, attr, np.array(value))
+    return func
 
-# Generate fluid systems
-def customize_strawberry_smoothie_material(mat):
-    # We modify the specular reflection color
-    mat.specular_reflection_color = np.array([1.0, 0.64, 0.64])
+
+class GranularSystem(PhysicalParticleSystem):
+    """
+    Particle system class simulating granular materials. Individual particles are composed of custom USD objects.
+    """
+
+    @classproperty
+    def is_fluid(cls):
+        return False
+
+    @classmethod
+    def _create_particle_prototypes(cls):
+        # Load the particle template
+        particle_template = cls._create_particle_template()
+        og.sim.import_object(obj=particle_template, register=False, auto_initialize=True)
+
+        assert len(particle_template.links) == 1, "GranularSystem particle template has more than one link"
```

Comment:
comment here? similar to macro particle system

Reply:
done!